tests/unit/metadata/test_multi_table.py

```diff
@@ -645,3 +645,109 @@ def test_detect_table_from_dataframe_table_already_exists(self):
+
+    def test__validate_table_exists(self):
+        """Test ``_validate_table_exists`` correctly raises an error."""
+        # Setup
+        metadata = MultiTableMetadata()
+        metadata._tables = {'table1': 'val', 'table2': 'val'}
+
+        # Run
+        metadata._validate_table_exists('table1')
+
+        # Assert
+        err_msg = "Unknown table name 'table3'."
+        with pytest.raises(InvalidMetadataError, match=err_msg):
+            metadata._validate_table_exists('table3')
+
+    def test_set_primary_key(self):
+        """Test ``set_primary_key``.
+
+        The method should validate the table exists and call
+        ``SingleTableMetadata.set_primary_key``.
+
+        Input:
+            - Table name
+            - Column name
+        """
+        # Setup
+        metadata = MultiTableMetadata()
+        metadata._tables = {'table1': Mock(), 'table2': 'val'}
+        metadata._validate_table_exists = Mock()
+
+        # Run
+        metadata.set_primary_key('table1', 'col')
+
+        # Assert
+        metadata._validate_table_exists.assert_called_once_with('table1')
+        metadata._tables['table1'].set_primary_key.assert_called_once_with('col')
+
+    def test_set_sequence_key(self):
+        """Test ``set_sequence_key``.
+
+        The method should validate the table exists and call
+        ``SingleTableMetadata.set_sequence_key``.
+
+        Input:
+            - Table name
+            - Column name
+        """
+        # Setup
+        metadata = MultiTableMetadata()
+        metadata._tables = {'table1': Mock(), 'table2': 'val'}
+        metadata._validate_table_exists = Mock()
+
+        # Run
+        warn_msg = 'Sequential modeling is not yet supported on SDV Multi Table models.'
+        with pytest.warns(match=warn_msg):
+            metadata.set_sequence_key('table1', 'col')
+
+        # Assert
+        metadata._validate_table_exists.assert_called_once_with('table1')
+        metadata._tables['table1'].set_sequence_key.assert_called_once_with('col')
+
+    def test_set_alternate_keys(self):
+        """Test ``set_alternate_keys``.
+
+        The method should validate the table exists and call
+        ``SingleTableMetadata.set_alternate_keys``.
+
+        Input:
+            - Table name
+            - List of column names
+        """
+        # Setup
+        metadata = MultiTableMetadata()
+        metadata._tables = {'table1': Mock(), 'table2': 'val'}
+        metadata._validate_table_exists = Mock()
+
+        # Run
+        metadata.set_alternate_keys('table1', ['col1', 'col2'])
+
+        # Assert
+        metadata._validate_table_exists.assert_called_once_with('table1')
+        metadata._tables['table1'].set_alternate_keys.assert_called_once_with(['col1', 'col2'])
+
+    def test_set_sequence_index(self):
+        """Test ``set_sequence_index``.
+
+        The method should validate the table exists and call
+        ``SingleTableMetadata.set_sequence_index``.
+
```

Comment:
For all of these can we just add a little bit in the Setup about what we're mocking. For example, in this case we're setting the `_tables` dict manually and mocking the tables